spec/controllers/needs_controller_spec.rb

```diff
@@ -0,0 +1,101 @@
+require 'rails_helper'
+require 'spec_helper'
+
+RSpec.describe NeedsController do
+
+  before(:each) do
+    controller.instance_variable_set(:@current_user, {})
+  end
+
+  let(:need) do
+    need = class_double("Need").as_stubbed_const
+    allow(need).to receive_messages(includes: need, page: need, order: need)
+    need
+  end
+
+  describe "GET #index" do
+    it "orders by created_at in descending order by default" do
+      expect(need).to receive(:order).with(:created_at => :desc).and_return(need)
+      get :index
+    end
+
+    it "filters on page number passed in params" do
+      expect(need).to receive(:page).with("5").and_return(need)
+      get :index, params: { page: 5 }
```

Comment:
could you add `expect(response).to be_successful` after the get action, it seems a bit odd we don't verify anything after the call